integration_test/lit.cfg.py

```diff
@@ -133,6 +133,17 @@
   config.substitutions.append(
       ('%ieee-sim', os.path.join(config.questa_path, "vsim")))
 
+# Enable Icarus Verilog as a fallback if no other ieee-sim was detected.
+if config.iverilog_path != "" and 'ieee-sim' not in config.available_features:
+  tool_dirs.append(os.path.dirname(config.iverilog_path))
+  tools.append('iverilog')
+  tools.append('vvp')
+  config.available_features.add('iverilog')
+  config.available_features.add('ieee-sim')
+  config.available_features.add('rtl-sim')
+  config.substitutions.append(('%iverilog', config.iverilog_path))
+  config.substitutions.append(('%ieee-sim', config.iverilog_path))
+
```

Comment:
I like that this adds a fallback option for `ieee-sim`. Would it make sense to move the `'ieee-sim' not in config.available_features` inside the `if` such that it only affects the `config.substitutions.append(('%ieee-sim', config.iverilog_path))` line? Otherwise, if someone had iverilog *and* questa installed, they wouldn't be able to run any iverilog-specific tests since the fact that questa already defines `ieee-sim` would skip this entire if statement.

Reply:
We definitely want to be able to write questa or iverilog, etc specific tests.

Reply:
Good idea, I'll change it.